### PR TITLE
Fix misaligned access

### PR DESCRIPTION
**Pull request by janisozaur**
Platforms like ARM are very sensitive to unaligned access. This change
makes sure the addresses obey the C++ standard and are properly aligned.

### CODE REVIEW

**Comment by IntelOrca** [comment]
It looks like we now always prefix `std::` for our memcpys and memsets.

File: src/openrct2/ride/Ride.h [hunk 1]
```diff
@@ -135,8 +137,6 @@ struct rct_ride_entry
     void* obj;
 };
 
-#pragma pack(pop)
-
```
Comment (IntelOrca) [inline src/openrct2/ride/Ride.h]: This change probably breaks testpaint.

Comment (janisozaur) [reply src/openrct2/ride/Ride.h]: This PR:
```
[  PASSED  ] 2847 tests.
[  FAILED  ] 202 tests, listed below:
```
develop:
```
[  PASSED  ] 2847 tests.
[  FAILED  ] 202 tests, listed below:
```

**Comment by janisozaur** [comment]
@IntelOrca re memcpy: the CIs pass, should I still prefix it?

**Comment by IntelOrca** [comment]
@janisozaur I think @ZehMatt said it guaranteed inlining or something, it was done for performance.

File: src/openrct2/ride/Ride.h (hunk 1 above)
Comment (IntelOrca) [reply src/openrct2/ride/Ride.h]: ok cool

**Comment by janisozaur** [comment]
@ZehMatt what macro? I updated the calls as requested.

**Pull request merged**